### PR TITLE
Use `ExpectedHypervolumeImprovement` candidates function for `BotorchSampler`

### LINKED ISSUE #4960: Add analytic `ExpectedHyperVolumeImprovement` candidate function to `BoTorchSampler`
### Motivation

Straightforward extension of #4546 to multi-objective version.

Quasi-Monte Carlo acquisition functions are for difficult cases where analytic value is not available, e.g. batch suggestion or considering running trials. When analytic form is available, it is better to use it.

### Description

Add a candidate function that uses [botorch.acquisition.multi_objective.analytic.ExpectedHypervolumeImprovement](https://botorch.org/api/acquisition.html#botorch.acquisition.multi_objective.analytic.ExpectedHypervolumeImprovement)

### Alternatives (optional)

_No response_

### Additional context (optional)

_No response_

### PR DESCRIPTION
**Pull request by adjeiv**
## Motivation
Resolve #4960 

## Description of the changes
- Add ehvi_candidates_func
- Use ehvi_candidates_func as default when we have multiple objectives, but no constraints and no running/pending trials
- Modify tests

Limitations:
- Constraints are not supported
- X_pending is not supported

### CODE REVIEW

File: optuna/integration/botorch.py [hunk 1]
```diff
@@ -592,7 +658,9 @@ def _get_default_candidates_func(
     ],
     "torch.Tensor",
 ]:
-    if n_objectives > 3:
+    if n_objectives > 3 and not has_constraint and not consider_running_trials:
+        return ehvi_candidates_func
+    elif n_objectives > 3:
```
Comment (adjeiv) [inline optuna/integration/botorch.py:663]: @contramundum53 is this the kind of condition we want to trigger using EHVI? Let me know, and I'll update tests accordingly.

Comment (contramundum53) [reply optuna/integration/botorch.py:663]: Yes, this is correct. Thanks!

**Comment by HideakiImamura** [comment]
@not522 @nabenabe0928 Could you review this PR?

**Comment by nabenabe0928** [comment]
> @not522 @nabenabe0928 Could you review this PR?

Sure")

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
Thank you for the PR!
I left some comments!

File: optuna/integration/botorch.py [hunk 2]
```diff
@@ -421,6 +422,71 @@ def qehvi_candidates_func(
...
+
+    train_y = train_obj
+    train_x = normalize(train_x, bounds=bounds)
+
+    model = SingleTaskGP(train_x, train_y, outcome_transform=Standardize(m=train_y.size(-1)))
+    mll = ExactMarginalLogLikelihood(model.likelihood, model)
+    fit_gpytorch_mll(mll)
+
+    # Approximate box decomposition similar to Ax when the number of objectives is large.
+    # https://github.com/facebook/Ax/blob/master/ax/models/torch/botorch_moo_defaults
```
[the remark this answers is not in the document]
Comment (nabenabe0928) [reply optuna/integration/botorch.py]: Could you also do the same for [here](https://github.com/optuna/optuna/pull/5065/files#diff-fb416f53f506e50d99de583a111470730758756ff9237bed4649f807491737aeR387-R388)?

**Review by not522 — COMMENTED** [review]
Thank you for your PR!
Could you update the document of `candidates_func`?
image

**Review by not522 — APPROVED** [review]
Thank you for your update. LGTM!

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

**Pull request merged**